pkgs/build-support/build-fhs-userenv/default.nix

```diff
@@ -1,35 +1,80 @@
-{ callPackage, runCommand, writeScript, stdenv, coreutils }:
+{ callPackage, runCommand, writeShellScriptBin, stdenv, coreutils, bubblewrap }:
 
 let buildFHSEnv = callPackage ./env.nix { }; in
 
-args@{ name, runScript ? "bash", extraInstallCommands ? "", meta ? {}, passthru ? {}, ... }:
+args @ {
+  name,
+  runScript ? "bash",
+  extraInstallCommands ? "",
+  meta ? {},
+  passthru ? {},
+  bwrap_args ? "",
+  automount ? true,
+  chdir ? "$(pwd)",
+  ...
+}:
 
+with builtins;
 let
-  env = buildFHSEnv (removeAttrs args [ "runScript" "extraInstallCommands" "meta" "passthru" ]);
+  env = buildFHSEnv (removeAttrs args [
+    "runScript" "extraInstallCommands" "meta" "passthru"
+    "bwrap_args" "automount" "chdir"
+  ]);
 
   chrootenv = callPackage ./chrootenv {};
 
-  init = run: writeScript "${name}-init" ''
-    #! ${stdenv.shell}
+  init = run: writeShellScriptBin "${name}-init" ''
     for i in ${env}/* /host/*; do
       path="/''${i##*/}"
       [ -e "$path" ] || ${coreutils}/bin/ln -s "$i" "$path"
```

Comment:
This can be removed I think, It should be handled by `ro_mounts` and `--ro-bind /etc /host/etc` right?